File: bukkit/src/main/java/me/devnatan/inventoryframework/logging/BukkitLogger.java
package me.devnatan.inventoryframework.logging;

import org.jetbrains.annotations.Nullable;

public final class BukkitLogger implements Logger {

    private final java.util.logging.Logger logger;
    private final String viewName;
    private final boolean isShaded;
    private String finalPrefix;

    public BukkitLogger(java.util.logging.Logger logger, String viewName, boolean isShaded) {
        this.logger = logger;
        this.viewName = viewName;
        this.isShaded = isShaded;
    }

    @Override
    @Nullable
    public String getPrefix() {
        if (finalPrefix == null) {
            finalPrefix = String.format("%s[%s]", isShaded ? "[IF]" : "", viewName) + " ";
        }

        return finalPrefix;
    }

    @Override
    public void debug(String message) {
        logger.info(getPrefix() + message);
    }

    @Override
    public void warn(String message) {
        logger.warning(getPrefix() + message);
    }

    @Override
    public void error(String message) {
        logger.severe(getPrefix() + message);
    }
}

File: inventory-framework-platform-minestom/src/main/kotlin/me/devnatan/inventoryframework/logging/BukkitLogger.kt
package me.devnatan.inventoryframework.logging

class BukkitLogger(
    private val logger: java.util.logging.Logger,
    private val viewName: String,
    private val isShaded: Boolean
) : Logger {
    private var finalPrefix: String? = null

    override fun getPrefix(): String? {
        if (finalPrefix == null) {
            finalPrefix = String.format("%s[%s]", if (isShaded) "[IF]" else "", viewName) + " "
        }

        return finalPrefix
    }

    override fun debug(message: String) {
        logger.info(prefix + message)
    }

    override fun warn(message: String) {
        logger.warning(prefix + message)
    }

    override fun error(message: String) {
        logger.severe(prefix + message)
    }
}
